### PR TITLE
Remove all references to old dm-verity boot tooling

### PR DESCRIPTION
**Pull request by dmcilvaney**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] If you are adding/removing a .spec file that has multiple-versions supported, please add [@microsoft/cbl-mariner-multi-package-reviewers](https://github.com/orgs/microsoft/teams/cbl-mariner-multi-package-reviewers) team as reviewer [(Eg. golang has 2 versions 1.18, 1.21+)](https://github.com/microsoft/azurelinux/tree/2.0/SPECS/golang)
- [x] Ready to merge

---

###### Summary 

**Breaking up PR into parts**:
- https://github.com/microsoft/azurelinux/pull/10973
- https://github.com/microsoft/azurelinux/pull/10974
- https://github.com/microsoft/azurelinux/pull/10975

Remove the now pointless parts of https://github.com/microsoft/azurelinux/pull/549 (since https://github.com/microsoft/azurelinux/pull/9537 removed dm-verity support from the imager tool).

Some parts of #549 were general refractors which are still needed, but everything related to building an image with dm-verity enabled has been removed.

See #9537 for details on migration paths using the imagecustomizer tool

###### Change Log  

- Remove `ReadOnlyVerityRoot` config struct
- Remove `verity-read-only-root` package
- Remove logic around verity roots in the toolkit

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- https://microsoft.visualstudio.com/DefaultCollection/OS/_workitems/edit/51951945

###### Test Methodology

- https://dev.azure.com/mariner-org/mariner/_build/results?buildId=670454&view=results

### CODE REVIEW

File: toolkit/docs/formats/imageconfig.md [hunk 1]
```diff
@@ -416,58 +416,9 @@ A sample KernelOptions specifying a default kernel:
...
-
-``` json
-"ReadOnlyVerityRoot": {
-    "Enable": true,
-    "TmpfsOverlays": [
-        "/var"
-    ],
-},
-```
+See [Azure Linux Image Customizer configuration](../../tools/imagecustomizer/docs/configuration.md#verity-type) for more information.
```
Comment (PawelWMS) [inline toolkit/docs/formats/imageconfig.md:421]: Is this a GA blocker as well? Technically, we shouldn't deprecate features after we GA. Or was that already considered and the risk is of breaking anyone is negligible?

Comment (dmcilvaney) [reply toolkit/docs/formats/imageconfig.md:421]: https://github.com/microsoft/azurelinux/pull/9537 should have disabled support already, I just forgot to remove this bit of the docs.

**Comment by dmcilvaney** [comment]
I've rebased this to the current state of the tools.

File: toolkit/docs/formats/imageconfig.md [hunk 2]
```diff
@@ -416,58 +416,9 @@ A sample KernelOptions specifying a default kernel:
 
 ### ReadOnlyVerityRoot
 
-"ReadOnlyVerityRoot" key controls making the root filesystem read-only using dm-verity.
-It will create a verity disk from the partition mounted at "/". The verity data is stored as
-part of the image's initramfs. More details can be found in [Misc: Read Only Roots](../how_it_works/5_misc.md#dm-verity-read-only-roots)
+ReadOnlyVerityRoot has been deprecated in the image configuration. The feature is suppored in the [Azure Linux Image Customizer](../../tools/imagecustomizer/README.md) tool now.
```
Comment (cwize1) [inline toolkit/docs/formats/imageconfig.md:419]: nit: `supported`

**Review by cwize1 — APPROVED** [review]

File: toolkit/docs/formats/imageconfig.md (hunk 2 above)
Comment (dmcilvaney) [reply toolkit/docs/formats/imageconfig.md:419]:
```suggestion
ReadOnlyVerityRoot has been deprecated in the image configuration. The feature is supported in the [Azure Linux Image Customizer](../../tools/imagecustomizer/README.md) tool now.
```

**Comment by dmcilvaney** [comment]
Parts merged in other prs

**Pull request closed without merging**